cmd/genji/shell/command.go

```diff
@@ -136,3 +140,132 @@ func displaySuggestions(in string) error {
 	fmt.Println()
 	return nil
 }
+
+// dumpTable display the content of the given table as SQL statements.
+func dumpTable(db *genji.DB, t *database.Table) error {
+	// Get table info.
+	ti, err := t.Info()
+	if err != nil {
+		return err
+	}
+
+	fields := ti.FieldConstraints
+	if len(fields) == 0 {
+		fmt.Printf("CREATE TABLE %s;\n", t.Name())
+	} else {
+		fmt.Printf("CREATE TABLE %s(", t.Name())
+		for i, f := range fields {
+			fmt.Printf("%s %s", fields[i].Path, strings.ToUpper(fields[i].Type.String()))
+			if f.IsPrimaryKey {
+				fmt.Printf(" PRIMARY KEY")
+			}
+
+			if f.IsNotNull {
+				fmt.Printf(" NOT NULL")
+			}
+
+			// Don't print the last comma.
+			if i == (len(fields) - 1) {
+				fmt.Println()
+				break
+			}
+
+			fmt.Println(",")
+		}
+
+		fmt.Println(");")
+	}
+
+	// Indexes section.
+	indexes, err := t.Indexes()
+	if err != nil {
+		return err
+	}
+
+	// Display indexes.
+	for _, index := range indexes {
+		u := ""
+		if index.Opts.Unique {
+			u = " UNIQUE"
+		}
+
+		info := fmt.Sprintf("CREATE%s INDEX %s ON %s (%s);", u, index.Opts.IndexName, index.Opts.TableName, index.Opts.Path)
+		fmt.Println(info)
+	}
+
+	// Inserts section.
+	return db.View(func(tx *genji.Tx) error {
+		q := fmt.Sprintf("SELECT * FROM %s", t.Name())
+		res, err := tx.Query(q)
+		if err != nil {
+			return err
+		}
+		defer res.Close()
+
+		c, err := res.Count()
```

Comment:
This will loop over all the documents unnecessarily. If the table is empty `res.Iterate` will simply return nil without calling the inner function, we don't need to have a special case here

Reply:
I was to defensive.